File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/turn/activity/SelectStoryTellerActivity.java
package fr.erban.dxitcompanion.game.turn.activity;

import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.view.View;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.res.ResourcesCompat;

import java.util.List;

import fr.erban.dxitcompanion.R;
import fr.erban.dxitcompanion.game.GameBean;
import fr.erban.dxitcompanion.game.player.PlayerBean;
import fr.erban.dxitcompanion.game.turn.Turn;

public class SelectStoryTellerActivity extends AppCompatActivity {

    private Turn turn;

    private GameBean gameBean;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.select_storyteller);

        turn = Turn.builder()
                .build();

        this.gameBean = (GameBean) getIntent().getSerializableExtra("Game");

        final RadioGroup radioGroup = findViewById(R.id.radioGroupChooseStoryteller);

        if (gameBean != null) {
            gameBean.setCurrentTurn(gameBean.getCurrentTurn() + 1);
            addPlayerNames(gameBean, radioGroup);
        }

        final TextView turnNumber = findViewById(R.id.turnNumber);
        final String turnNumberComplete = getString(R.string.turnNumberPrefix) + gameBean.getCurrentTurn();
        turnNumber.setText(turnNumberComplete);

    }

    private void addPlayerNames(GameBean gameBean, RadioGroup radioGroup) {

        final List<PlayerBean> players = gameBean.getPlayers();

        for (int i = 0, playersSize = players.size(); i < playersSize; i++) {
            PlayerBean player = players.get(i);
            RadioButton radioButton = new RadioButton(this);
            radioButton.setText(player.getName());
            radioButton.setVisibility(View.VISIBLE);
            radioButton.setTextSize(50);
            final Typeface font = ResourcesCompat.getFont(this, R.font.write_me_a_song);
            radioButton.setTypeface(font);
            radioButton.setTextColor(getResources().getColor(R.color.backgroundTextColor));

            radioGroup.addView(radioButton);

            if (i == 0) {
                radioButton.setChecked(true);
            }
        }
    }

    public void continueToTurnResults(View view) {

        final RadioGroup radioGroup = findViewById(R.id.radioGroupChooseStoryteller);

        final int selectedPlayerId = radioGroup.getCheckedRadioButtonId();
        final View rb = radioGroup.findViewById(selectedPlayerId);
        int idx = radioGroup.indexOfChild(rb);
        final RadioButton radioButton = (RadioButton) radioGroup.getChildAt(idx);
        final String storyTellerName = radioButton.getText().toString();

        final List<PlayerBean> players = gameBean.getPlayers();

        for (PlayerBean player : players) {
            if (player.getName()
                    .equals(storyTellerName)) {
                turn.setStoryTeller(player);
            }
        }

        Intent intent = new Intent(SelectStoryTellerActivity.this, EveryoneFoundActivity.class);
        intent.putExtra("Game", gameBean);
        intent.putExtra("Turn", turn);
        SelectStoryTellerActivity.this.startActivity(intent);
    }
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/turn/activity/SelectStoryTellerActivity.kt
package fr.erban.dxitcompanion.game.turn.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.RadioButton
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.res.ResourcesCompat
import fr.erban.dxitcompanion.R
import fr.erban.dxitcompanion.common.parcelableExtra
import fr.erban.dxitcompanion.databinding.SelectStorytellerBinding
import fr.erban.dxitcompanion.game.GameBean
import fr.erban.dxitcompanion.game.turn.Turn

class SelectStoryTellerActivity : AppCompatActivity() {

    private lateinit var binding: SelectStorytellerBinding
    private lateinit var gameBean: GameBean
    private var turn = Turn()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = SelectStorytellerBinding.inflate(layoutInflater)
        setContentView(binding.root)

        gameBean = intent.parcelableExtra<GameBean>("Game")!!
            .copy(currentTurn = intent.parcelableExtra<GameBean>("Game")!!.currentTurn + 1)

        binding.turnNumber.text = getString(R.string.turnNumberPrefix) + gameBean.currentTurn

        gameBean.players.forEachIndexed { index, player ->
            val radioButton = RadioButton(this).apply {
                text = player.name
                visibility = View.VISIBLE
                textSize = 50f
                typeface = ResourcesCompat.getFont(this@SelectStoryTellerActivity, R.font.write_me_a_song)
                setTextColor(resources.getColor(R.color.backgroundTextColor, theme))
            }
            binding.radioGroupChooseStoryteller.addView(radioButton)
            if (index == 0) radioButton.isChecked = true
        }
    }

    fun continueToTurnResults(view: View) {
        val selectedId = binding.radioGroupChooseStoryteller.checkedRadioButtonId
        val rb = binding.radioGroupChooseStoryteller.findViewById<RadioButton>(selectedId)
        val storyTellerName = rb.text.toString()

        val storyTeller = gameBean.players.first { it.name == storyTellerName }
        turn = turn.copy(storyTeller = storyTeller)

        startActivity(
            Intent(this, EveryoneFoundActivity::class.java)
                .putExtra("Game", gameBean)
                .putExtra("Turn", turn)
        )
    }
}
